app/src/main/java/ru/thstdio/aa2020/api/Repository.kt

```diff
@@ -1,51 +1,96 @@
+import kotlinx.coroutines.async
+import kotlinx.coroutines.coroutineScope
+import kotlinx.serialization.ExperimentalSerializationApi
 import kotlinx.serialization.json.Json
-import ru.thstdio.aa2020.api.response.*
-import ru.thstdio.aa2020.data.Actor
-import ru.thstdio.aa2020.data.Genre
-import ru.thstdio.aa2020.data.Movie
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import ru.thstdio.aa2020.api.converter.toActor
+import ru.thstdio.aa2020.api.converter.toCinema
+import ru.thstdio.aa2020.api.converter.toCinemaDetail
+import ru.thstdio.aa2020.api.interceptor.ApiHeaderInterceptor
+import ru.thstdio.aa2020.api.response.ConfigurationResponse
+import ru.thstdio.aa2020.api.response.Genre
+import ru.thstdio.aa2020.api.service.TimeDbApi
+import ru.thstdio.aa2020.data.Cinema
+import ru.thstdio.aa2020.data.CinemaDetail
 
-class Repository(private val appContext: Context) {
-    private val jsonFormat = Json { ignoreUnknownKeys = true }
+private const val API_KEY =
+    "eyJhbGciOiJIUzI1NiJ9.eyJhdWQiOiIwZjUzODY0YjlmMTViNzE1OWM0YWNlMjE4NWQyNmY2MyIsInN1YiI6IjVlNmExMWE5MzU3YzAwMDAxMzNlZjdiNyIsInNjb3BlcyI6WyJhcGlfcmVhZCJdLCJ2ZXJzaW9uIjoxfQ.4dMlOhOdXjKDjvEHUwAuZDpNBc_xZaEi0z3YMLUqV8w"
+private const val BASE_URL = "https://api.themoviedb.org/3/"
 
-    suspend fun downloadMovies(): List<Movie> {
-        val moviesJson = loadFromJson<List<JsonMovie>>("data.json", Json::decodeFromString)
+@ExperimentalSerializationApi
+class Repository() {
+    private val api: TimeDbApi = createApi()
 
-        val actors = loadActors().associateBy { it.id }
-        val genres = loadGenres().associateBy { it.id }
-        return moviesJson.map { it.toMovie(actors, genres) }
-    }
+    private var configuration: ConfigurationResponse? = null
+    private var genres: Map<Long, Genre>? = null
 
-    suspend fun downloadMovie(id: Long): Movie =
-        withContext(Dispatchers.IO) {
-            delay(1400)
-            downloadMovies().first { it.id == id }
+    private fun createApi(): TimeDbApi {
+        val json = Json {
+            ignoreUnknownKeys = true
         }
+        val client = OkHttpClient().newBuilder()
+            .addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BASIC))
+            .addInterceptor(ApiHeaderInterceptor(API_KEY))
+            .build()
+        val retrofit: Retrofit = Retrofit.Builder()
+            .client(client)
+            .baseUrl(BASE_URL)
+            .addConverterFactory(json.asConverterFactory("application/json".toMediaType()))
+            .build()
+        return retrofit.create(TimeDbApi::class.java)
+    }
 
 
-    private suspend fun loadGenres(): List<Genre> =
-        loadFromJson<List<JsonGenre>>("genres.json", Json::decodeFromString)
-            .map { it.toGenre() }
+    suspend fun downloadMovies(page: Int): Pair<List<Cinema>, Int> = coroutineScope {
```

Comment:
what u think about public api of this class?
Pair<List<Cinema>, Int> will be clear for new developer on project?